cupyx/scipy/special/_basic.py

```diff
@@ -0,0 +1,19 @@
+from cupy import _core
+
+
+log1p = _core.create_ufunc(
+    "cupyx_scipy_log1p",
+    ("f->f", "d->d"),
+    "out0 = out0_type(log1p(in0));",
```

Comment:
We can support also complex type inputs.
https://github.com/cupy/cupy/blob/master/cupy/_core/include/cupy/complex.cuh#L43

Reply:
```suggestion
    (('f->f', 'out0 = log1pf(in0)'), "d->d', 'F->F', 'D->D'),
    "out0 = log1p(in0);",
```
How about using `log1pf` for float32 inputs?

Reply:
Thanks, I have updated to use that.

For complex-valued inputs, this does work okay, but the implementation in SciPy is not simply `log(x + 1)`. For reference the `clog1p` implementation that gets used in the complex case is  [defined here](https://github.com/scipy/scipy/blob/8f640fd997e0386bdd4525a03869c45c726ac3cc/scipy/special/_cunity.pxd#L25-L66). There is even one branch there that even uses long double precision. 

With a simple `F->F` as suggested above, the logspace tests were failing for me with `complex64` outside of `logspace(-2.5, 2.5)` whereas with double precision it passes until close to `logspace(-12, 12)`. For that reason, I currently have it cast to complex128 for complex64 inputs. I guess `log1pf` is doing something more numerically accurate than `logf(1 + x)` internally so the `float32` case didn't have the same accuracy issue.